runtime/convertValues_test.go

```diff
@@ -2961,6 +3365,335 @@ func TestRuntimeStringValueImport(t *testing.T) {
 	})
 }
 
+func TestTypeValueImport(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("Type<Int>", func(t *testing.T) {
+
+		t.Parallel()
+
+		typeValue := cadence.NewTypeValue(cadence.IntType{})
+
+		script := `
+            pub fun main(s: Type) {
+                log(s.identifier)
+            }
+        `
+
+		encodedArg, err := json.Encode(typeValue)
+		require.NoError(t, err)
+
+		rt := NewInterpreterRuntime()
+
+		var ok bool
+
+		runtimeInterface := &testRuntimeInterface{
+			decodeArgument: func(b []byte, t cadence.Type) (value cadence.Value, err error) {
+				return json.Decode(b)
+			},
+			log: func(s string) {
+				assert.Equal(t, s, "\"Int\"")
+				ok = true
+			},
+		}
+
+		_, err = rt.ExecuteScript(
+			Script{
+				Source:    []byte(script),
+				Arguments: [][]byte{encodedArg},
+			},
+			Context{
+				Interface: runtimeInterface,
+				Location:  TestLocation,
+			},
+		)
+
+		require.NoError(t, err)
+		require.True(t, ok)
+	})
+
+	t.Run("missing struct", func(t *testing.T) {
+
+		t.Parallel()
+
+		typeValue := cadence.NewTypeValue(&cadence.StructType{
+			QualifiedIdentifier: "S",
+			Location:            TestLocation,
+			Fields:              []cadence.Field{},
+			Initializers:        [][]cadence.Parameter{},
+		})
+
+		script := `
+            pub fun main(s: Type) {
+            }
+        `
+
+		encodedArg, err := json.Encode(typeValue)
+		require.NoError(t, err)
+
+		rt := NewInterpreterRuntime()
+
+		runtimeInterface := &testRuntimeInterface{
+			decodeArgument: func(b []byte, t cadence.Type) (value cadence.Value, err error) {
+				return json.Decode(b)
+			},
+		}
+
+		_, err = rt.ExecuteScript(
+			Script{
+				Source:    []byte(script),
+				Arguments: [][]byte{encodedArg},
+			},
+			Context{
+				Interface: runtimeInterface,
+				Location:  TestLocation,
+			},
+		)
+
+		require.Error(t, err)
```

Comment:
Looks like the last commit reverted the assertion suggestion again: [`bec485a` (#1202)](https://github.com/onflow/cadence/pull/1202/commits/bec485ae1d813bf91c260c231067ad19fd6b5c8f#diff-863660c79fae3d9b381c2dff3025cd88dc46eb62571b754e7695acbfeab4fbfeL3454-R3454)